packages/cli/src/deploy/ensureFunctions.ts

```diff
@@ -40,23 +42,45 @@ export async function ensureFunctions({
     toAdd.map((func) => {
       const { namespace } = hexToResource(func.systemId);
       if (namespace === "") {
-        return writeContract(client, {
-          chain: client.chain ?? null,
-          address: worldDeploy.address,
-          abi: worldAbi,
-          // TODO: replace with batchCall (https://github.com/latticexyz/mud/issues/1645)
-          functionName: "registerRootFunctionSelector",
-          args: [func.systemId, func.systemFunctionSignature, func.systemFunctionSelector],
-        });
+        return pRetry(
+          () =>
+            writeContract(client, {
+              chain: client.chain ?? null,
+              address: worldDeploy.address,
+              abi: worldAbi,
+              // TODO: replace with batchCall (https://github.com/latticexyz/mud/issues/1645)
+              functionName: "registerRootFunctionSelector",
+              args: [func.systemId, func.systemFunctionSignature, func.systemFunctionSelector],
+            }),
```

Comment:
we could potentially expand `writeContract`'s API to take in retry options or do finer grained retries (e.g. just retry simulation), but I want to think more about that before deviating from viem